src/cozycode/capstone/parking/ParkingGarage.java

```diff
@@ -68,4 +73,128 @@ public static void main(String[] args) {
+            }
+            ParkingSpace thisSpace = freeList.get(0);
+            thisSpace.assign(car);
+            serializeToFile(spaces, "parkingData.txt");
+            return thisSpace.getPosition();
+        }
+
+        // for EVs
+        else if (Car.carType() == CarType.EV){
+            // creates new list of applicable spots
+            for (int i = 0; i < spaces.length; i ++) {
+                for (int j = 0; j < spaces[i].length; j++) {
+                    if (space.getType() == CarType.EV) {
+                        typeList.add(space);
+                    }
+                }
+            }
+            // creates new list of applicable spots that are open
+            for (ArrayList[] space: typeList){
+                if(ParkingSpace.isAvailable == true){
+                    freeList.add(space);
+                }
+            }
+            ParkingSpace thisSpace = freeList.get(0);
+            thisSpace.assign(car);
+            serializeToFile(spaces, "parkingData.txt");
+            return thisSpace.getPosition();
+        }
+
+        // for carpools
+        else if (Car.carType() == CarType.CARPOOL){
+            // creates new list of applicable spots
+            for (int i = 0; i < spaces.length; i ++) {
+                for (int j = 0; j < spaces[i].length; j++) {
+                    if (space.getType() == CarType.CARPOOL) {
+                        typeList.add(space);
+                    }
+                }
+            }
+            // creates new list of applicable spots that are open
+            for (ArrayList[] space: typeList){
+                if(ParkingSpace.isAvailable == true){
+                    freeList.add(space);
+                }
+            }
+            ParkingSpace thisSpace = freeList.get(0);
+            thisSpace.assign(car);
+            serializeToFile(spaces, "parkingData.txt");
+            return thisSpace.getPosition();
+        }
+
+        // for regulars and above types if specialty spots run out
+        else{
+            if(spaces.length == 0){
+                return -1;
+            }
+            else{
+                // creates new list of applicable spots
+                for (int i = 0; i < spaces.length; i ++) {
+                    for (int j = 0; j < spaces[i].length; j++) {
+                        if (space.getType() == CarType.REGULAR) {
+                            typeList.add(space);
+                        }
+                    }
+                }
+                // creates new list of applicable spots that are open
+                for (ArrayList[] space: typeList){
+                    if(ParkingSpace.isAvailable == true){
+                        freeList.add(space);
+                    }
+                }
+                ParkingSpace thisSpace = freeList.get(0);
+                thisSpace.assign(car);
+                serializeToFile(spaces, "parkingData.txt");
+                return thisSpace.getPosition();
+            }
```

Comment:
What if there are no regular spaces available? You checked if there are no spaces at all, not if there are no empty spaces. If there are no available spaces, to my knowledge this would error.